src/features/UpdatePowerShell.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import fetch from "node-fetch";
+import { compare, parse, prerelease, SemVer } from "semver";
+import { MessageItem, window } from "vscode";
+import { LanguageClient } from "vscode-languageclient";
+import Settings = require("../settings");
+import { EvaluateRequestType } from "./Console";
+
+const PowerShellGitHubReleasesUrl =
+        "https://api.github.com/repos/PowerShell/PowerShell/releases/latest";
+const PowerShellGitHubPrereleasesUrl =
+    "https://api.github.com/repos/PowerShell/PowerShell/releases";
+
+export class GitHubReleaseInformation {
+    public static async FetchLatestRelease(preview: boolean): Promise<GitHubReleaseInformation> {
+        // Fetch the latest PowerShell releases from GitHub.
+        let releaseJson: any;
+        if (preview) {
+            // This gets all releases and the first one is the latest prerelease if
+            // there is a prerelease version.
+            releaseJson = (await fetch(PowerShellGitHubPrereleasesUrl)
+                .then((res) => res.json())).find((release: any) => release.prerelease);
+        } else {
+            releaseJson = await fetch(PowerShellGitHubReleasesUrl)
+                .then((res) => res.json());
+        }
+
+        return new GitHubReleaseInformation(
+            releaseJson.tag_name, releaseJson.assets);
+    }
+
+    public version: SemVer;
+    public isPreview: boolean = false;
+    public assets: any[];
+
+    public constructor(version: string | SemVer, assets: any[] = []) {
+        this.version = parse(version);
+
+        if (prerelease(this.version)) {
+            this.isPreview = true;
+        }
+
+        this.assets = assets;
+    }
+}
+
+interface IUpdateMessageItem extends MessageItem {
+    id: number;
+}
+
+export async function InvokePowerShellUpdateCheck(
+    languageServerClient: LanguageClient,
+    localVersion: SemVer,
+    arch: string,
+    release: GitHubReleaseInformation) {
+    const options: IUpdateMessageItem[] = [
+        {
+            id: 0,
+            title: "Yes!",
```

Comment:
Some people don't like 'shouting' in messages and in terms of UX it is usually advised against.
```suggestion
            title: "Yes",
```